higher-order.js

```diff
@@ -3,6 +3,46 @@
  * The root object of the library containing the higher-order functions.
  */
 const _ = {
+    /**
+     * Given an array and a reducer function, returns the single value from calling 
+     * reducer on each element of the array and accumulating the result.
+     * @param {Array} arr 
+     * @param {Function} reducer 
+     * @param {Any} initialValue 
+     */
+    reduce: function(arr, reducer, initialValue) {
+        if (typeof arr != 'object' || !(arr instanceof Array)) {
```

Comment:
Will `typeof arr != 'object'` ever be `false` but `!(arr instanceof Array)` be `true`?  There might be a case, but I can't think of it.  If there is, there should be a test that captures this branch.